src/include/sof/common.h

```diff
@@ -63,5 +63,12 @@
 #define IS_ENABLED_STEP_2(values) IS_ENABLED_STEP_3(values 1, 0)
 #define IS_ENABLED_STEP_3(ignore, value, ...) (!!(value))
 
+/* GCC 9.x has checks for pointer alignment within packed structures when
+ * cast from one type to another. This macro should only be used after checking
+ * alignment. This compiler builtin may not be available on all compilers so
+ * this macro can be defined as NULL if needed.
+ */
+#define ASSUME_ALIGNED(x, a) __builtin_assume_aligned(x, a)
+
```

Comment:
@lgirdwood Could you check compilation with Xtensa C Compiler ? I believe a conditional compilation of ASSUME_ALIGNED() macro has to be added.

Reply:
Thanks @lbetlej, I was kind of expecting that. I guess the xcc wont support but maybe the llvm update will.